### DISCUSSION TITLE
Ideas for .valetrc file

### DISCUSSION THREAD

**Discussion by mikeybinns** [laravel/valet#1389] [Ideas]
I have some ideas on how to improve the `.valetrc` file. Currently there's no real reason to switch because it does the same as the existing `.valetphprc` file, however I do believe in its potential.

Allow shorter syntax for the PHP version:
`php=8.1` instead of `php=php@8.1` for berevity

Add option to define sitename:
`sitename="application"`

Add option to define isolation:
`isolate=true`
`isolate=false`

Add option to define secure:
`secure=true`
`secure=false`

Add option to define proxies:
`proxy="http://127.0.0.1:9200"`

Allow definition of site location:
This would allow the `.valetrc` file to be in a different location than the site root. 
Some example use cases:
I have a WordPress project where I'm working on the theme in the themes folder, but my valetrc must be in the WordPress root, meaning i must go up the directory tree to run commands. Instead, I could have the valetrc file with a property like so:
`site_location="../../../"`

I am building a site into a subdirectory of my project root, meaning I must cd into the public folder to execute valet commands.
 Instead, I could have the valetrc file with a property like so:
`site_location="./public"`

Site envs:
referring to [Site Specific Environment Variables](https://laravel.com/docs/10.x/valet#site-specific-environment-variables)
Set `$_SERVER['key']` to `value` for this specific site like so:
`ENV.KEY="value"`

If we implemented this, then we could possibly define everything in a git committed file, allowing new developers to join a project by just running `valet link` for example

Here's a full example for a potential `.valetrc` file:
```
sitename="application"
php=8.1
isolate=true
secure=true
proxy="http://127.0.0.1:9200"
site_location="./public"
ENV.KEY="value"
```

**Comment by AlexanderWright** [comment]
How about:

XDEBUG_STRING="PHPSTORM"
XDEBUG_PORT=12321
etc.

**Reply by mikeybinns** [reply]
I think those options would fall under ENV, as they aren't used by valet (as far as I know) but by the site?
So you'd say:
```
ENV.XDEBUG_STRING="PHPSTORM"
ENV.XDEBUG_PORT=12321
```

**Comment by PierW** [comment]
For me another idea is to write the composer version. For any project with php 7.1 maybe you should use composer 1 and not 2 (an error will be show you in the terminal)
I would like to have something: composer=1 or composer=2 and it will do the rollback automatically.

**Comment by mikeybinns** [comment]
@mattstauffer I just wanted to let you know about this as you wrote the original PR to introduce valetrc 🙂

**Reply by mattstauffer** [reply]
@mikeybinns I don't get it.. This *just* showed up in my notifications because of the comment below, but not because of your original tag. This is a great list! Thank you so much!!

**Reply by mikeybinns** [reply]
No worries, glad you like it 😄

**Comment by dgualvez** [comment]
hi, this sounds like a good feature.. is this already available?
